Tests implementation

martin-re opened the pull request.

dav1d-b commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,149 @@
+                'rateLimitFrequencyInSeconds': 5,
+                'ownerEmail': 'fm-base-workers',
+            },
+        ]
```
Why don't we test this with just 1 task definition? The implementations are almost the same, I don't see a reason for testing the same thing twice.

dav1d-b commented on tests/common_test/workflow_test/test_task_generator.py:
```diff
@@ -0,0 +1,1016 @@
+        assert test_mock == test_task
+
+    def test_dyn_fork_arrays_def_task(self) -> None:
+        task_inputs = InventoryWorkflows.InstallDeviceByName.WorkflowInput()
```
I thought we agreed that we will not be using workflow definitions e.g. InstallDeviceByName that were implemented in the old version of the SDK. 

1. Tests are also form of code documentation.
2. InstallDeviceByName has execute methods with parameter `task: Task`, which has recently changed to `worker_input: WorkerInput`
3. We are providing wrong examples for future users of this library in these tests (We provide wrong documentation).

We should not just copy old tests mindlessly, but we should review if the old code is usable / good and then change it if it is necessary.

daviddzxy commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,114 @@
+class TestTaskGenerator:
+    def test_create_task_def(self) -> None:
+        class Http(ServiceWorkersImpl):
+            class HttpTask1(WorkerImpl):
```
```suggestion
            class HttpTask(WorkerImpl):
```

dav1d-b commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,114 @@
+
+class TestTaskGenerator:
+    def test_create_task_def(self) -> None:
+        class Http(ServiceWorkersImpl):
```
I don't think we need to create ServiceWorkersImpl Class just to test the WorkerImpl behaviour. Why dont we test it like in the test_create_task_def_with_custom_default test case?

dav1d-b commented on tests/common_test/workflow_test/test_task_generator.py:
```diff
@@ -0,0 +1,1016 @@
+
+    def test_decision_case_value_task(self) -> None:
+        test_task = task.DecisionCaseValueTask(
+            type = TaskType.DECISION,
```
```suggestion
            type=TaskType.DECISION,
```

dav1d-b commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,114 @@
+                    status_code: int = Field(..., alias='statusCode')
+                    cookies: dict[str, Any]
+
+                def execute(self, task_def: Task) -> TaskResult:
```
```suggestion
                def execute(self, worker_input: WorkerInput) -> TaskResult:
```

dav1d-b commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,114 @@
+                status_code: int = Field(..., alias='statusCode')
+                cookies: dict[str, Any]
+
+            def execute(self, task: Task) -> TaskResult:
```
```suggestion
            def execute(self, worker_input: WorkerInput) -> TaskResult:
```

dav1d-b commented on tests/conftest.py:
```diff
@@ -0,0 +1,38 @@
+
+    class WorkerOutput(TaskOutput): ...
+
+    def execute(self, task: Task) -> TaskResult:
```
```suggestion
    def execute(self, worker_input: WorkerInput) -> TaskResult:
```

dav1d-b commented on tests/conftest.py:
```diff
@@ -0,0 +1,38 @@
+
+    class WorkflowOutput(WorkflowImpl.WorkflowOutput): ...
+
+    def workflow_builder(self, workflow_inputs: Any) -> None: ...
```
Why workflow_inputs: **Any** and not workflow_inputs: WorkflowInput?

Jozefiel commented on .github/workflows/pr-check.yml:
```diff
@@ -48,3 +48,6 @@ jobs:
       - name: Run pyright
         run: |
           poetry run pyright .
+      - name: Run tests
```
Use a more specific name. We will run more tests in the future

Commit:
some typing fixes 2 (mypy)

Commit:
tests added
- use pytest instead of unittest
- old code moved to conftest to keep clean repo
- call tests in pre-check CI
- fix problem with pythonpath to run tests

Commit:
some typing fixes (mypy)

Commit:
use mocks instead of old code from fm-base-workers
- review fixes

Commit:
fix mixedCase in tests (ruff)

Commit:
some typing fixes 3 (mypy)

Commit:
review fixes

Commit:
update typehints

Commit:
(ruf) some conventions fixes

Commit:
fix path to tests in pyproject

dav1d-b commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,108 @@
+            class WorkerDefinition(TaskDefinition):
+                name: str = 'HTTP_task'
+                description: str = 'Generic http task'
+                labels: ListAny = ['BASIC', 'HTTP']
```
```suggestion
                labels: list[str] = ['BASIC', 'HTTP']
```

Jozefiel replied:
list[str] is invalid because we also allow an empty list. Can be fixed in another PR

dav1d-b commented on tests/common_test/worker_test/test_task_def.py:
```diff
@@ -0,0 +1,108 @@
+            class WorkerDefinition(TaskDefinition):
+                name: str = 'HTTP_task'
+                description: str = 'Generic http task'
+                labels: ListAny = ['BASIC', 'HTTP']
```
```suggestion
                labels: list[str] = ['BASIC', 'HTTP']
```

Jozefiel replied:
list[str] is invalid because we also allow an empty list. Can be fixed in another PR

Commit:
review fixes
